src/Symfony/Component/DependencyInjection/ContainerBuilder.php

```diff
@@ -1187,31 +1214,34 @@ private function doResolveServices(mixed $value, array &$inlineServices = [], bo
             $reference = $value->getValues()[0];
             $value = fn () => $this->resolveServices($reference);
         } elseif ($value instanceof IteratorArgument) {
-            $value = new RewindableGenerator(function () use ($value, &$inlineServices) {
+            $containerRef = \WeakReference::create($this);
```

Comment:
This seems to include a bunch of changes that are unrelated to generating lazy closures AFAICT

Reply:
moved to #49639